File: src/0828-Unique-Letter-String/0828.js
var uniqueLetterString = function(S) {
    let res = 0, f = 0, mod = 1000000007;
    let m = new Array(26), n = new Array(26);
    m.fill(0), n.fill(0);
    for (let i = 0; i < S.length; i++) {
        let t = S.charCodeAt(i) - 65;
        f = f + 1 + i - 2* m[t] + n[t];
        res = (res + f) % mod;
        n[t] = m[t];
        m[t] = i + 1;
    }
    return res;
};

File: src/0828-Unique-Letter-String/0828.py
class Solution:
    def uniqueLetterString(self, S: str) -> int:
        res, mod, f = 0, 10**9 + 7, 0
        m, n = [0]*26, [0]*26
        for i, v in enumerate(S):
            t = ord(v) - 65
            f = f + 1 + i - 2 * m[t] + n[t]
            res = (res + f) % mod
            n[t], m[t] = m[t], i + 1
        return res